packages/web-client/app/routes/application.ts

```diff
@@ -1,5 +1,25 @@
 import Route from '@ember/routing/route';
+import RouterService from '@ember/routing/router-service';
+import { LocationService } from '../services/location';
+import { inject as service } from '@ember/service';
+import '@cardstack/web-client/css/variables.css';
+import config from '@cardstack/web-client/config/environment';
 
-import '../css/variables.css';
+export default class ApplicationRoute extends Route {
+  @service declare router: RouterService;
+  @service declare location: LocationService;
 
-export default class ApplicationRoute extends Route {}
+  beforeModel() {
+    // if we are on the card space domain, we should not allow
+    // users to visit other routes besides index
+    // some important functionality will be messed up
+    // eg. layer 1 infura calls via WalletConnect
+
+    // NOTE: there is a caveat to this. If the user uses a LinkTo from this page
+    // that leads to, say, card-pay.wallet, this interception will not run
+    // it may be better to create a base class for all routes that has this functionality
+    if (this.location.hostname.endsWith(config.cardSpaceHostnameSuffix)) {
+      this.router.replaceWith('index');
+    }
+  }
```

Comment:
The redirect implemented here prevents folks from visiting child routes when visiting x.card.space.... Not sure how to handle this caveat, or if we should think about this caveat for now? Would lean towards not thinking about it.

There is a known way to prevent transitions:

https://guides.emberjs.com/v3.28.0/routing/preventing-and-retrying-transitions/

But I'm not very confident in handling this event correctly.

Reply:
yes, let’s not think about it for now 😆